tar_scm.py

```diff
@@ -261,6 +261,9 @@ def detect_version(self, args, repodir):
         def get_timestamp(self, args, repodir):
             d = {"parent_tag": None, "versionformat": "%ct"}
             timestamp = self.detect_version(d, repodir)
+            # avoid future times due to broken commits
+            if datetime.datetime.fromtimestamp(timestamp) > datetime.datetime.now():
+                  timestamp = datetime.datetime.now()
             return int(timestamp)
```

Comment:
I would prefer that an exception is raised in this case. Otherwise the diff-mechanism of obscpio will detect changes on  every call of service until the broken timestamp is in the past.